internal/provider/resource_gitlab_pages_domain.go

```diff
@@ -0,0 +1,220 @@
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"auto_ssl_enabled": {
+				Description: "Enables automatic generation of SSL certificates issued by Let’s Encrypt for custom domains.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"certificate": {
+				Description: "The certificate in PEM format with intermediates following in most specific to least specific order.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"key": {
+				Description: "The certificate key in PEM format.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Default:     nil,
+			},
+			"url": {
+				Description: "The URL for the given domain.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"certificate_data": {
+				Description: "The certificate data.",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Computed:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"expired": {
+							Description: "Is the certificate expired?",
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Optional:    true,
+						},
+						"expiration": {
+							Description:  "The certificate expiration date.",
+							Type:         schema.TypeString,
+							ValidateFunc: validation.IsRFC3339Time,
+							Computed:     true,
+							Optional:     true,
+						},
+					},
+				},
+			},
+			"verified": {
+				Description: "The certificate data.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"verification_code": {
+				Description: "The verification code for the domain.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+})
+
+func resourceGitlabPagesDomainCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	projectId := d.Get("project").(string)
+	domain := d.Get("domain").(string)
+	auto_ssl_enabled := d.Get("auto_ssl_enabled").(bool)
+	certificate := d.Get("certificate").(string)
+	key := d.Get("key").(string)
+
+	options := &gitlab.CreatePagesDomainOptions{
+		Domain:         &domain,
+		AutoSslEnabled: &auto_ssl_enabled,
+		Certificate:    &certificate,
+		Key:            &key,
+	}
+
+	log.Printf("[DEBUG] create gitlab pages domain %s", domain)
+
+	_, _, err := client.PagesDomains.CreatePagesDomain(projectId, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(buildTwoPartID(&projectId, &domain))
+	return resourceGitlabPagesDomainRead(ctx, d, meta)
+}
+
+func resourceGitlabPagesDomainUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	projectID := d.Get("project").(string)
```

Comment:
It's a common practise to use the `id` here as well, like you did for the `Read()`